### ISSUE TITLE
`NullPointerException` in `publishBuildInfo` when not run on a node

### ISSUE THREAD

**Issue by TBBle** [jfrog/jenkins-artifactory-plugin#471]
**Describe the bug**
In a scripted pipeline, simply ending with `artifactory.publishBuildInfo` (where `artifactory` is returned from `Artifactory.server` ) causes a `java.lang.NullPointerException`:

```
java.lang.NullPointerException
	at org.jfrog.hudson.pipeline.common.Utils.extractVcs(Utils.java:127)
	at org.jfrog.hudson.pipeline.common.executors.PublishBuildInfoExecutor.execute(PublishBuildInfoExecutor.java:33)
	at org.jfrog.hudson.pipeline.scripted.steps.PublishBuildInfoStep$Execution.run(PublishBuildInfoStep.java:45)
	at org.jfrog.hudson.pipeline.scripted.steps.PublishBuildInfoStep$Execution.run(PublishBuildInfoStep.java:33)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**To Reproduce**

I can't publish the whole Pipeline script, but the shape is
```Groovy
import org.jfrog.hudson.pipeline.common.types.ArtifactoryServer
import org.jfrog.hudson.pipeline.common.types.buildInfo.BuildInfo
import org.jfrog.hudson.pipeline.common.types.Docker
 
// See https://github.com/jfrog/jenkins-artifactory-plugin/blob/master/src/main/java/org/jfrog/hudson/pipeline/scripted/dsl/ArtifactoryPipelineGlobal.java
// for source of the Artifactory global object.
 
// Used to gather information for metadata on uploads to Artifactory
ArtifactoryServer artifactory = Artifactory.server 'our-server'
BuildInfo buildInfo = Artifactory.newBuildInfo()
Docker artifactoryDocker = Artifactory.docker server: artifactory

parallel X: { node('docker') {
    // docker build, etc.
    BuildInfo localBuildInfo = artifactoryDocker.push "${serverRepository}:${p4Changelist}", rtDockerRepository
    buildInfo.append localBuildInfo
}}, Y: { node('somenode') {
    // Build and package artefacts, etc.
    BuildInfo localBuildInfo = artifactory.upload failNoOp: true, spec: uploadSpec
    buildInfo.append localBuildInfo
}}

artifactory.publishBuildInfo buildInfo
```

The uploads themselves all work fine, and have the relevant build properties (`build.timestamp`, etc.).

When processing the `artifactory.publishBuildInfo buildInfo` line, the above exception occurs.

In case it's relevant, the VCS here is Perforce, not Git.

The problem goes away if I change the last line to
```Groovy
stage('Publish BuildInfo') {
   node('somerandomnode') {
        artifactory.publishBuildInfo buildInfo
    }
}
```

**Expected behavior**

No `NullPointerException`, and the build passes.

**Screenshots**

N/A

**Versions**
- Jenkins Artifactory plugin version: 3.9.0
- Jenkins operating system: 2.222.1
- Artifactory Version: 6.11.3

**Additional context**

I suspect this is a symptom of #213, as the [relevant line](https://github.com/jfrog/jenkins-artifactory-plugin/blob/artifactory-3.9.0/src/main/java/org/jfrog/hudson/pipeline/common/Utils.java#L127) is the `return filePath.act` in
```Java
    public static Vcs extractVcs(FilePath filePath, Log log) throws IOException, InterruptedException {
        return filePath.act(new MasterToSlaveFileCallable<Vcs>() {
            public Vcs invoke(File f, VirtualChannel channel) throws IOException {
                return GitUtils.extractVcs(f, log);
            }
        });
    }
```

**Comment by bverkron** [comment]
We ran into what I think is this same issue.  Same error from same step, but only on RHEL8 based jenkins nodes it seems. Our Jenkins agents are containerized and anything that was based on RHEL8 or ubi8 base images had this problem. RHEL7 based images did not.

Today I ran the same job with the same agent image on our jenkins dev master and it worked.  The difference was our dev server had been upgraded to version 3.11.3 of the plugin. I downgraded back to our previous version on that server (3.10.6) and it started throwing the nullexception error like our primary servers running the RHEL8 agents.

Not sure what has changed (didn't dig deeply into this repo, just found this bug report). Might be worth a try @TBBle

**Comment by Or-Geva** [comment]
@TBBle, It may help to upgrade the plugin to version 3.11.3 as @bverkron suggests.
Looking forward to your feedback.

**Comment by TBBle** [comment]
Interesting; we use CentOS 7-based build nodes, and (I think) a CentOS 7-based master. It certainly wouldn't be RHEL 8, anyway.

I'll flag the plugin upgrade to the team maintaining the Jenkins instance, we have a few plugins that need upgrading already, but this is a little complicated as a combination of bugs means we need to do a core Jenkins upgrade before I can test Jenkinsfile changes in Perforce without bringing the whole CI system down in a screaming pile of build engineers, and I never verified if this reproduced with Git, as our Git-based pipelines are only now migrating from `sh curl ...` to `rtUpload`.

**Comment by eyalbe4** [comment]
This issue is fixed and released as part of version 3.13.1 of the Jenkins Artifactory Plugin. We would appreciate any feedback for the fix.
